File: linear_search.py
def take_input(array):
	#this function takes input from the user and stores it in the array
    nums = int(input("Enter the number of elements in the array: "))
    for i in range(nums):
        element = int(input())
        array.append(element)


def display(array):
	#this function is responsible for printing the array
    print(array)


def search_element(array):
	#this function is responsible for the linear sort mechanism
    to_be_searched = int(input("Enter the element to be found: "))
    index_of_element = 0
    isFound = False
    for i in range(len(array)):
        if to_be_searched == array[i]:
            index_of_element = i
            isFound = True
            break

    if isFound:
        print("The element has been found at index", index_of_element)
    else:
        print("The element is not present in the array")


if __name__ == "__main__":
    array = []
    take_input(array)
    display(array)
    search_element(array)

File: linear_search.cpp
#include <iostream>
// we will be using vectors for this because vectors can dynamically
// allocate memory and can improve the overrall performance of the code
#include <vector>
using namespace std;

void takeInput(vector<int> &v)
// this function is responsible for taking input of the array
{
    // we allow the user to add as many elements as they wish to
    cout << "Keep typing elements till you want" << endl;
    cout << "Type 0 to stop adding elements to the array" << endl;
    int input;
    // whenever they type the number 0, the array will stop taking input 
    while(cin >> input && input != 0)
    {
        // adding elements to the array
        v.push_back(input);
    }
}

void printArray(vector<int> &v)
// this is responsible for printing the array
{

    for (auto &element : v)
    // this is a range based loop
    {
        cout << element << " ";
    }
    cout << endl;    
}


void linearSearch(vector<int> &v)
// this is the function that is responsible for the linear srearch funcitonality
{
    int toBeFound, indexFound;
    bool found = false;
    cout << "Enter the element to be searhed for in the array: " << endl;
    cin >> toBeFound;
    // linear search functionality
    for (int i = 0; i < v.size(); i++)
    {
        if (v[i] == toBeFound)
        {
            indexFound = i;
            found = true;
            break;
        }
    }

    if (found)
    {
        cout << "The element has been found at index position: " << indexFound;
        cout << endl;
    } 
    else 
    {
        cout << "The element is not present in the array" << endl;
    }
}


int main()
{
    // the driver code for the program
    vector<int> numbers;
    takeInput(numbers);
    printArray(numbers);
    linearSearch(numbers);
    return 0;
}